### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by viveshok** [moka-rs/moka#64]
Hi team -- thanks for making `moka` available to the world!

I'm having some difficulty using it, for me the cache simply grow till it run out of memory.

Below is a minimum reproducible example. It grows to a few gigs of memory usage after a few minutes. Am I not incantating the cache properly? (thanks for your patience i'm a Rust n00b)

```rust
use moka::unsync::{Cache, CacheBuilder};
use ahash::RandomState;

use std::time::Duration;
use rand::Rng;

const CACHE_SIZE: u64 = 1_000_000;
const TTI: Duration = Duration::from_millis(10_000);

fn main() {
    let mut cache: Cache<(u64, u64, u64), (), RandomState> = CacheBuilder::new(CACHE_SIZE as usize)
        .initial_capacity(CACHE_SIZE as usize)
        .time_to_idle(TTI)
        .build_with_hasher(RandomState::default());

    let mut rng = rand::thread_rng();
    loop {
        let key = (rng.gen::<u64>(), rng.gen::<u64>(), rng.gen::<u64>());
        cache.insert(key, ());
    }
}
```

**Comment by tatsuya6502** [comment]
Thank you for reporting the issue with the reproducer! You are using Moka in a correct way. I found Moka is leaking memory in eviction/expiration queues when expiring cached entries.

I believe I just fixed it on a new topic branch. You can give it a try by replacing the dependency in `Cargo.toml` with the following:

**Cargo.toml**

```toml
moka = { git = "https://github.com/moka-rs/moka.git", branch = "fix-deq-mem-leak" }
```

I will run some pre-release tests on it later today (it is 8:50 AM in my timezone UTC+0800), and if everything goes well, I will release Moka v0.7.1 with the fix.

----

Also, I think you are currently using Moka v0.6.3 or earlier. If so, you will need to change this:

**src/main.rs**

```rust
... = CacheBuilder::new(CACHE_SIZE as usize)
```
to one of the followings as v0.7.0 now takes the `max_capacity` as `u64`.

```rust
... = CacheBuilder::new(CACHE_SIZE)
```

or

```rust
... = Cache::builder()
    .max_capacity(CACHE_SIZE)
```

Note that you do not have to change this: `initial_capacity(CACHE_SIZE as usize)`. That method still takes `usize`.

Thanks!

**Comment by tatsuya6502** [comment]
FYI, we will apply the same fix to Moka v0.7 and v0.6 lines:

- v0.7: #65
- v0.6: #66
